src/sentry/search/moresnuba/backend.py

```diff
@@ -0,0 +1,318 @@
+from __future__ import absolute_import
+
+import datetime
+import functools
+import time
+from datetime import timedelta
+from hashlib import md5
+
+from django.db.models import Q
+from django.utils import timezone
+
+from sentry import quotas
+from sentry.api.event_search import convert_search_filter_to_snuba_query, TAG_KEY_RE
+from sentry.utils import snuba, metrics
+from sentry.search.snuba.base import (
+    SnubaSearchBackendBase,
+    QCallbackCondition,
+    QuerySetBuilder,
+    assigned_to_filter,
+    unassigned_filter,
+    logger,
+    dependency_aggregations,
+    paginator_results_estimator,
+)
+
+sort_strategies = {
+    "date": "events.last_seen",
+    "freq": "times_seen",
+    "new": "events.first_seen",
+    "priority": "priority",
+}
+
+aggregation_defs = {
+    "times_seen": ["count()", ""],
+    "events.first_seen": ["multiply(toUInt64(min(events.timestamp)), 1000)", ""],
+    "events.last_seen": ["multiply(toUInt64(max(events.timestamp)), 1000)", ""],
+    # https://github.com/getsentry/sentry/blob/804c85100d0003cfdda91701911f21ed5f66f67c/src/sentry/event_manager.py#L241-L271
+    "priority": ["toUInt64(plus(multiply(log(times_seen), 600), events.last_seen))", ""],
+    # Only makes sense with WITH TOTALS, returns 1 for an individual group.
+    "events.total": ["uniq", "events.issue"],
+}
+
+issue_only_fields = set(["query", "bookmarked_by", "assigned_to", "unassigned", "subscribed_by"])
+
+
+class MoreSnubaSearchBackend(SnubaSearchBackendBase):
```

Comment:
Let's give this backend a self explanatory name and a docstring that explains why it exists.
Written this way it would be very hard for someone reading the code to understand the design.